fix: transaction hash computation (#1303)
starknet-io/starknet-docs: components/Starknet/modules/architecture-and-concepts/pages/network-architecture/messaging-mechanism.adoc

[source,cairo]
----
l1_handler_tx_hash = _h_(
    "l1_handler",
    version,
    contract_address,
    entry_point_selector,
    _h_(calldata),

>     0,

Reviewer: ```suggestion
    0, // L1 to L2 message fees are charged on the L1.
```

Reviewer 2: @glihm Good clarification. I moved this to the list of explanations below.

----

> l1_handler_tx_hash = ℎ(

Reviewer 3: ```suggestion
l1_handler_tx_hash = _h_(
```

    "l1_handler",

>     version, // at the moment only version 0 is supported

Reviewer 3: ```suggestion
    version, // at the moment, only version 0 is supported
```

    contract_address,
    entry_point_selector,

>     ℎ(calldata),

Reviewer 3: ```suggestion
    _h_(calldata),
```

Where:

- `l1_handler` is a constant prefix, encoded in bytes (ASCII), as big-endian.

> - `version` is the transaction version. Only version 0 is currently supported.

Reviewer: We may refer to the new table to have this being true forever, even if a new version shows up?

- `l1_handler` is a constant prefix, encoded in bytes (ASCII), as big-endian.
- `chain_id` is a constant value that specifies the network to which this transaction is sent.

> - _h_ is the xref:architecture-and-concepts:cryptography/hash-functions.adoc#pedersen_hash[Pedersen] hash (note that since we're hashing an array, the # of inputs needs to be appended to the hash).

Reviewer 3: ```suggestion
- _h_ is the xref:architecture-and-concepts:cryptography/hash-functions.adoc#pedersen_hash[Pedersen] hash (note that since we are hashing an array, the number of inputs needs to be appended to the hash).
```

Reviewer: If it's a serialized array, `appended` may be misleading. The length of the array must be the first element of the array to be hashed.
`[len, c1, c2, c3, ...]`.

Reviewer 2: @glihm Good catch. How about this:

```suggestion
- _h_ is the xref:architecture-and-concepts:cryptography/hash-functions.adoc#pedersen_hash[Pedersen] hash. Be aware that because an array is hashed, the length of the array must be the first element on the array.
```
